### PR TITLE
feat(masthead-2.1): update mobile layout & styles

### PR DESCRIPTION
**Pull request by andy-blum**
### Related Ticket(s)

[ADCMS-2612](https://jsw.ibm.com/browse/ADCMS-2612)

Followup to #9829 

### Description

Updates masthead left-nav to use new masthead v2.1 styles & data structure

### Changelog

**Changed**

- Updates to style of left-nav and sub-components

### CODE REVIEW

**Comment by jkaeser** [comment]
@andy-blum I've been keeping this initiative's work isolated in `feat/masthead-v2.1`. Can you merge it into this branch and set it as the PR's target branch?

If you find that level of isolation unnecessary, we can instead merge `feat/masthead-v2.1` into `feat/masthead-v2-dev`, then merge `feat/masthead-v2-dev` back into this branch again.

File: packages/web-components/src/components/masthead/left-nav-menu-section.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ import settings from 'carbon-components/es/globals/js/settings.js';
 import HostListener from 'carbon-web-components/es/globals/decorators/host-listener.js';
 import HostListenerMixin from 'carbon-web-components/es/globals/mixins/host-listener.js';
 import ChevronLeft20 from 'carbon-web-components/es/icons/chevron--left/20.js';
-import ArrowRight20 from 'carbon-web-components/es/icons/arrow--right/20.js';
+// import ArrowRight20 from 'carbon-web-components/es/icons/arrow--right/20.js';
```
Comment (jkaeser) [inline packages/web-components/src/components/masthead/left-nav-menu-section.ts:15]: Nit: remove commented code

File: packages/web-components/src/components/masthead/masthead-composite.ts [hunk 2]
```diff
@@ -345,6 +345,19 @@ class DDSMastheadComposite extends HostListenerMixin(LitElement) {
         `;
       }
 
+      if (elem.isHeading) {
```
Comment (jkaeser) [inline packages/web-components/src/components/masthead/masthead-composite.ts:348]: Do we need this check, or can we add `.isHeading="${elem.isHeading}"` to the returned template fragment that follows this conditional instead?

**Review by jkaeser — CHANGES_REQUESTED** [review]
@andy-blum Looking good so far!

## Code Notes
1. We need to make sure the "selected" nav items still work. This is the mechanism that applies a special style to links that match the current page URL.

## Design Notes
1. It looks like the link text is still using `body-short-02`, but it should use `body-short-01`.
   Screen Shot 2023-01-25 at 12 17 46 PM
2. There's extra space beneath some headings. Pretty sure this is unwanted.
   - Screen Shot 2023-01-25 at 12 18 19 PM
   - Screen Shot 2023-01-25 at 12 24 27 PM
3. The "view all" CTAs should be blue and have an arrow. Or are you waiting for the specific ticket that handles view all (which would be fine, just want to check)?
   Screen Shot 2023-01-25 at 12 22 39 PM

**Comment by oliviaflory** [comment]
Will leave a few comments to help start some design QA cc @jwitkowski79 – would love your eyes too!

## 1 Title not vertically centered
Echoing John's observation in item 2: the menu title is not vertically centered. See `Faceted megamenu`, `Simple megamenu` as examples.
Screenshot 2023-02-03 at 3 34 49 PM

**Expecting** to be vertically centered in a 48px container (Example "Products)
Screenshot 2023-02-03 at 3 40 19 PM

## 2 No change
The extra space below the Heading 1 is correct, no changes needed 👍 
Screenshot 2023-02-03 at 3 44 09 PM

## 3 Focus state for titles
Focus state for menu title might be coming from the system/browser default? 
Screenshot 2023-02-03 at 3 46 41 PM


**Expecting** state to match desktop, see [Figma spec sheet](https://www.figma.com/file/35JfN0Sg4ltIIjM7jBDKMk/Navigation?node-id=3188%3A178305&t=HfLw8W7bSB8d2Rdy-1)
Screenshot 2023-02-03 at 3 48 08 PM

## 4 Title has extra vertical space (check with Jen)
Looks like the description might be holding extra space in the Heading 1? If no description is present I would have expected the container to be the line height of the heading size.
See `Faceted megamenu` / `No heading descriptions` as an example
Screenshot 2023-02-03 at 3 52 14 PM

Screenshot 2023-02-03 at 3 51 48 PM

## 5 Type styles
Echoing John's comment that the following items should be using `body-short-01` paired with a 16x16 px icon 
- back button
- menu title
- link item
Screenshot 2023-02-03 at 3 57 24 PM
Screenshot 2023-02-03 at 3 57 34 PM

**Review by jwitkowski79 — CHANGES_REQUESTED** [review]
@andy-blum Hello! A few other things not mentioned by @oliviaflory and @jkaeser.

### 1 Arrow alignment on clickable heading level 2
Currently the arrows are left aligned to the text. 
27BEB7A2-EC87-4470-9FA6-B98965EDE8C4



**Expecting** that they should be right aligned as seen in the Figma file [here ](https://www.figma.com/file/35JfN0Sg4ltIIjM7jBDKMk/Navigation?node-id=4415%3A202956&t=GNmMs67BSPb7kCl0-1) and in the below screenshot

5E27C5F1-694C-4B95-9E1D-2C843069F5C9


### 2 Title has extra vertical space
@oliviaflory is correct in stating that the heading level 1 title is holding extra vertical space if the description is not included. If no description is being used, the space should collapse and only the title spacing + line-height should be visible.

File: packages/web-components/src/components/masthead/masthead.scss [hunk 3]
```diff
@@ -10,7 +10,8 @@
 @import '@carbon/ibmdotcom-styles/scss/globals/vars';
 
 @import '@carbon/ibmdotcom-styles/scss/components/masthead/masthead';
-@import '@carbon/ibmdotcom-styles/scss/components/masthead/masthead-leftnav';
+// @import '@carbon/ibmdotcom-styles/scss/components/masthead/masthead-leftnav';
+@import '../../../../styles/scss/components/masthead/masthead-leftnav';
```
Comment (jkaeser) [inline packages/web-components/src/components/masthead/masthead.scss:14]: Oops!

File: packages/web-components/src/components/masthead/masthead-composite.ts [hunk 4]
```diff
@@ -345,26 +345,14 @@ class DDSMastheadComposite extends HostListenerMixin(LitElement) {
...
-      }
-
       return html`
         <dds-left-nav-menu-item
           ?last-highlighted=${elem.lastHighlightedItem}
           ?active="${elem.selected}"
           href="${elem.url}"
           title="${elem.title}"
           data-autoid="${elem.autoid}"
+          .isHeading=${elem.isHeading ?? false}
```
Comment (jkaeser) [inline packages/web-components/src/components/masthead/masthead-composite.ts:355]: 👏

**Review by jkaeser — CHANGES_REQUESTED** [review]
@andy-blum This is looking great! Almost there. Some dev changes got committed, and group headers should be type productive `heading-01`:

Screen Shot 2023-02-07 at 4 48 09 PM

**Review by jkaeser — APPROVED** [review]

**Comment by proeung** [comment]
@andy-blum - @jkaeser and I had a chance to chat today and we decided that it's best to merge these changes down so that we have all of the latest code in the `masthead-v2.1` branch. We can then create a deploy preview that both @oliviaflory and @jwitkowski79 can use to conduct a comprehensive design QA of the Masthead v2.1 updates.

**Pull request merged**